<a href="https://colab.research.google.com/github/Evgeniia1996/repository_one/blob/main/%D0%94%D0%97_19.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
df = pd.read_csv('real-data.csv')

In [ ]:
df.info()

In [41]:
df.describe()

female_life_expectancy  male_life_expectancy  \
count              191.000000            191.000000   
mean                74.076440             68.678010   
std                  7.793487              7.608412   
min                 53.100000             50.400000   
25%                 68.550000             63.200000   
50%                 75.200000             68.400000   
75%                 79.700000             74.150000   
max                 88.300000             83.200000   

       expected_schooling_female  expected_schooling_male  \
count                 187.000000               187.000000   
mean                   13.770053                13.397326   
std                     3.283023                 2.671261   
min                     4.500000                 6.600000   
25%                    11.450000                11.750000   
50%                    14.000000                13.400000   
75%                    16.000000                15.250000   
max                    21.800000                20.300000   

       female_mean_schooling  male_mean_schooling  female_gni_per_capita  \
count             179.000000           179.000000             178.000000   
mean                8.788268             9.323464           14622.404494   
std                 3.489555             2.937305           15826.774673   
min                 1.300000             2.200000             176.000000   
25%                 5.800000             6.900000            2894.500000   
50%                 9.700000             9.500000            8128.500000   
75%                11.700000            11.800000           22631.750000   
max                13.900000            14.300000           75094.000000   

       male_gni_per_capita  adolescent_birth_rate  female_parliament_seats  \
count           178.000000             191.000000               189.000000   
mean          24251.865169              44.484817                24.769312   
std           23875.373787              38.185359                12.466031   
min             797.000000               1.600000                 0.000000   
25%            6121.250000              11.100000                16.500000   
50%           15953.500000              36.200000                23.600000   
75%           36503.250000              64.150000                33.600000   
max          105348.000000             170.500000                55.700000   

       female_secondary_education  male_secondary_education  \
count                  177.000000                177.000000   
mean                    62.706780                 67.068362   
std                     29.528487                 26.450033   
min                      6.400000                 13.000000   
25%                     37.700000                 44.800000   
50%                     69.900000                 71.200000   
75%                     90.200000                 92.500000   
max                    100.000000                100.000000   

       female_labor_force  male_labor_force  
count           178.00000        178.000000  
mean             50.23764         69.900562  
std              15.71221          8.814937  
min               6.00000         43.900000  
25%              42.47500         65.100000  
50%              52.15000         69.350000  
75%              60.37500         75.475000  
max              83.10000         95.500000

In [ ]:
x=df.duplicated()

In [ ]:
x.describe()

дубликатов нет

In [ ]:
df1 = df.loc[:30]
df1

In [ ]:
df2 = df.loc[31:60]
df2

In [ ]:
result1 = pd.merge(df1, df2, on='female_life_expectancy', how='inner')
result1

In [ ]:
result2 = pd.merge(df1, df2, on='male_labor_force', how='left')
result2

In [22]:
from sklearn.preprocessing import normalize

In [ ]:
normalize_data = normalize(df, norm='max', axis=1)
normalize_df = pd.DataFrame(normalize_data, columns = df.columns)

не смогла провести нормализацию, почему не получается - не понимаю, убирала столбцы со строковыми значениями, все равно не получается

Всего строк 191. Максимальное количество пропусков - 14. Максимальный процент пропусков - 7. Небольшой процент, поэтому просто удалю строки с пустыми значениями.

In [43]:
df.dropna()

country  female_life_expectancy  \
0                           Afghanistan                    65.3   
1                               Albania                    79.2   
2                               Algeria                    78.0   
4                                Angola                    64.3   
6                             Argentina                    78.6   
..                                  ...                     ...   
186  Venezuela (Bolivarian Republic of)                    75.2   
187                            Viet Nam                    78.2   
188                               Yemen                    67.1   
189                              Zambia                    63.9   
190                            Zimbabwe                    62.0   

     male_life_expectancy  expected_schooling_female  expected_schooling_male  \
0                    58.9                        7.7                     12.7   
1                    74.1                       15.3                     13.7   
2                    74.9                       15.3                     14.0   
4                    59.0                       11.5                     12.9   
6                    72.2                       19.2                     16.6   
..                    ...                        ...                      ...   
186                  66.3                       13.8                     11.8   
187                  69.1                       13.2                     12.7   
188                  60.6                        7.7                     10.5   
189                  58.5                       10.9                     11.0   
190                  56.2                       12.0                     12.3   

     female_mean_schooling  male_mean_schooling  female_gni_per_capita  \
0                      2.3                  3.4                  533.0   
1                     11.7                 10.9                11637.0   
2                      7.7                  8.4                 3550.0   
4                      4.2                  6.9                 4751.0   
6                     11.4                 10.9                15581.0   
..                     ...                  ...                    ...   
186                   11.4                 10.8                 2866.0   
187                    8.0                  8.7                 6932.0   
188                    2.9                  5.1                  176.0   
189                    7.2                  7.2                 2615.0   
190                    8.3                  9.2                 3286.0   

     male_gni_per_capita maternal_mortality_ratio  adolescent_birth_rate  \
0                 3089.0                      638                   82.6   
1                16630.0                       15                   14.5   
2                17787.0                      112                   11.7   
4                 6197.0                      241                  138.4   
6                26376.0                       39                   39.1   
..                   ...                      ...                    ...   
186               6796.0                      125                   82.7   
187               8826.0                       43                   34.6   
188               2428.0                      164                   54.4   
189               3837.0                      213                  117.0   
190               4397.0                      458                   94.3   

     female_parliament_seats  female_secondary_education  \
0                       27.2                         6.4   
1                       35.7                        95.4   
2                        7.5                        46.0   
4                       29.5                        28.2   
6                       44.4                        71.0   
..                       ...                         ...   
186                     22.2    

После удаления осталось 169 строк